File: problems/leetcode/53.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        arr = []
        arr.append(nums[0])
        maxsum = arr[0]
        for i in range(1, len(nums)):
            arr.append(max(arr[i-1] + nums[i], nums[i]))
            if arr[i] > maxsum:
                maxsum=arr[i]
        return maxsum

File: problems/leetcode/53.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int c_sum = nums[0];
        int max_sum = nums[0];
        
        for (int i = 1; i<nums.size(); i++) {
            c_sum = max(nums[i], c_sum + nums[i]);
            max_sum = max(c_sum, max_sum);
        }
        return max_sum;

    }
};